puppet/lib/puppet/util/puppetdb/http.rb

```diff
@@ -3,14 +3,28 @@
 require 'net/http'
 require 'timeout'
 require 'pp'
+require 'thread'
+require 'puppet/util/puppetdb/atom'
 
 module Puppet::Util::Puppetdb
   class Http
+    SERVER_URL_FAIL_MSG = "Failing over to the next PuppetDB server_url in the 'server_urls' list"
 
-    SERVER_URL_FAIL_MSG = "Failing over to the next PuppetDB url in the 'server_urls' list"
+    if RUBY_PLATFORM == 'java'
+      begin
+        @@last_good_query_server_url_index = com::puppetlabs::puppetserver::PuppetDBTerminusState.getSharedStateAtom()
```

Comment:
so if you're running a PuppetDB with a Java puppet-server, but doesn't happen to include ^^^ it's going to throw a class not found sort of exception and assign a "local" Atom for that instance, is that how I'm reading this?